libs/base/ui/structure/text/src/text.component.ts

```diff
@@ -10,27 +10,52 @@ import { truncateFix } from './prehydrate';
 import { TextProperties } from './text.model';
 import { textStyles } from './text.styles';
 
-@hydratable(['mouseover', 'focusin'])
+@ssrShim('style')
+@hydratable()
 export class TextComponent extends LitElement implements TextProperties {
   static override styles = textStyles;
 
-  protected resizeObserver?: ResizeObserver;
-
   @property({ type: Boolean })
   hideToggle = false;
   @property({ type: Boolean, reflect: true })
   truncated = false;
   @property({ type: Boolean })
   defaultExpanded?: boolean;
-
   @property({ type: Boolean, reflect: true })
   truncation = false;
-
+  @property({ type: Number, reflect: true })
+  truncateAfter?: number;
   @property() readMoreLabel = 'Read more';
 
+  protected resizeObserver?: ResizeObserver;
+
+  protected willUpdate(changedProperties: PropertyValues): void {
+    if (changedProperties.has('truncateAfter')) {
+      if (this.truncateAfter) {
+        this.style.setProperty('--line-clamp', String(this.truncateAfter));
+      } else {
+        this.style.removeProperty('--line-clamp');
+      }
+    }
```

Comment:
nitpick: we could simplify as `this.style.removeProperty('--line-clamp', null);` should remove the property afaik